crates/bevy_ecs/src/storage/blob_vec.rs

```diff
@@ -119,6 +119,27 @@ impl BlobVec {
         }
     }
 
+    /// Reserves the minimum capacity for at least `additional` more elements to be inserted in the given `BlobVec`.
+    #[inline]
+    pub fn reserve(&mut self, additional: usize) {
+        if self.capacity - self.len < additional {
+            self.do_reserve(additional);
+        }
+    }
+
+    /// Similar to `reserve_exact`. This method ensures that the capacity will grow at least `self.capacity()` if there is no
+    /// enough space to hold `additional` more elements.
+    #[cold]
+    fn do_reserve(&mut self, additional: usize) {
+        let available_space = self.capacity - self.len;
+        if available_space < additional && self.item_layout.size() > 0 {
```

Comment:
If we are already doing the checks in reserve, we don't need to be repeating them here. We can just assume we have already met the requisite conditions.